functions/Find-DbaSimilarTable.ps1

```diff
@@ -0,0 +1,247 @@
+        $everyservervwcount = 0
+    }
+    process {
+        foreach ($Instance in $SqlInstance) {
+            try {
+                Write-Message -Level Verbose -Message "Connecting to $Instance"
+                $server = Connect-SqlInstance -SqlInstance $Instance -SqlCredential $SqlCredential
+            }
+            catch {
+                Write-Message -Level Warning -Message "Failed to connect to: $Instance"
+                continue
+            }
+
+            if ($IncludeSystemDatabases) {
+                $dbs = $server.Databases | Where-Object { $_.Status -eq "normal" }
+            }
+            else {
+                $dbs = $server.Databases | Where-Object { $_.Status -eq "normal" -and $_.IsSystemObject -eq $false }
+            }
+
+            if ($Database) {
+                $dbs = $dbs | Where-Object Name -In $Database
+            }
+
+            if ($ExcludeDatabase) {
+                $dbs = $dbs | Where-Object Name -NotIn $ExcludeDatabase
+            }
+
+            if ($ExcludeViews) {
+                $sql = $sql -replace '--<<TableTypeCondition>>--', " AND (t.TABLE_TYPE <> 'VIEW' AND t2.TABLE_TYPE <> 'VIEW') "
+            }
+
+            if ($SchemaName) {
+                $sql = $sql -replace '--<<SchemaNameCondition>>--', `
+                                        (" AND (c.TABLE_SCHEMA = '{0}') " -f $SchemaName.Replace("'","''"))  #Replace single quotes with two single quotes!                                     
+            }
+
+            if ($TableName) {
+                $sql = $sql -replace '--<<TableNameCondition>>--', `
+                                        (" AND (c.TABLE_NAME = '{0}') " -f $TableName.Replace("'","''"))     #Replace single quotes with two single quotes!                                     
+            }
+
+            if ($MatchPercentThreshold) {
+                $sql = $sql -replace '--<<MatchPercentThreshold>>--', `
+                                        (" AND (100 * COUNT(c2.COLUMN_NAME) / MIN(ColCountsByTable.Column_Count) >= '{0}') " -f $MatchPercentThreshold)
+            }
+
+            Write-Message -Level Debug -Message $sql
+
+            $totalcount = 0
+            $dbcount = $dbs.count
+            foreach ($db in $dbs) {
+                
+                Write-Message -Level Verbose -Message "Searching on database $db"
+                    
+                $props = 'ComputerName', 'InstanceName', 'SqlInstance', 'DatabaseName', 'MatchPercent', 'TableNameRankInDB', 'SchemaName', 'TableName', 'TableType', 'ColumnCount', 'MatchingDatabaseName', 'MatchingSchemaName', 'MatchingTableName', 'MatchingTableType', 'MatchingColumnCount'
+
+                $rows = $db.ExecuteWithResults($sql).Tables.Rows
+
+                Select-DefaultView -InputObject $rows -Property $props
+
+                $vwcount = $vwcount + $rows.Count
+                $totalcount = $totalcount + $rows.Count
+                $everyservervwcount = $everyservervwcount + $rows.Count
```

Comment:
For formatting purpose we use camelCase for variables in the functions. So `$everyservervwcount` would be written as `everyServerVwCount`.